src/Microsoft.Dnx.Compilation/BuildLoadContext.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Runtime.Loader;
+
+namespace Microsoft.Dnx.Compilation
+{
+    // This load context isn't tracked in the dictionary, it's a shim
+    // so that we don't allocte a real load context until absolutely necessary
+    internal class BuildLoadContext : LoadContext, IAssemblyLoadContext
+    {
+        private readonly Project _project;
+        private readonly CompilationEngine _compilationEngine;
+        private readonly CompilationEngineContext _compilationEngineContext;
+        private RuntimeLoadContext _projectLoadContext;
+        private readonly object _syncObject = new object();
+
+        public BuildLoadContext(Project project,
+                                CompilationEngine compilationEngine,
+                                CompilationEngineContext compilationEngineContext)
+        {
+            _project = project;
+            _compilationEngine = compilationEngine;
+            _compilationEngineContext = compilationEngineContext;
+        }
+
+        private RuntimeLoadContext LoadContext
+        {
+            get
+            {
+                lock (_syncObject)
+                {
+                    if (_projectLoadContext == null)
+                    {
+                        _projectLoadContext = InitializeProjectLoadContext();
+                    }
+                }
+
+                return _projectLoadContext;
+            }
+        }
+
+        Assembly IAssemblyLoadContext.Load(AssemblyName assemblyName)
+        {
+            try
+            {
+                return _compilationEngineContext.DefaultLoadContext.Load(assemblyName);
+            }
+            catch (FileNotFoundException)
+            {
+                return LoadContext.LoadWithoutDefault(assemblyName);
+            }
+        }
+
+        public override Assembly LoadAssembly(AssemblyName assemblyName)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
Nope. This should never be called. The base class only calls this via assembly resolve. This is a pass through implementation.